react/usearray_en.md

```diff
@@ -1,4 +1,43 @@
+Requirements
+- `removeAtIndex` method
+- `push` method
 
-There is no solution yet.
+Now, the implementation of the above methods are standard problems with the notable exception that we must update the array immutably each time. This necessity arises due to the requirement of re-rendering the component wherever this hook is utilized. React employs a strict equality check for optimization purposes, as explained in this [article](https://react.dev/learn/updating-objects-in-state), to differentiate state changes.
 
-Would you like to [contribute to the solution](https://github.com/BFEdev/BFE.dev-solutions/blob/main/react/usearray_en.md)? [Contribute guideline](https://github.com/BFEdev/BFE.dev-solutions#how-to-contribute)
+```
+// Solution by [Ankit Kumar](https://github.com/mynameisankit)
+
+import { useState, useCallback } from 'react'
+
+type UseArrayActions<T> = {
+  push: (item: T) => void,
+  removeByIndex: (index: number) => void
+}
+
+function removeAtIndex(index: number, array: any[], setArray: () => void): void {
+  const partBeforeIndex = array.slice(0, index) || [];
+  const partAfterIndex = array.slice(index + 1) || [];
+
+  const updatedArray = [...partBeforeIndex, ...partAfterIndex];
+  setArray(updatedArray);
+}
+
+function pushValue(value: any, array: any[], setArray: () => void): void {
```

Comment:
seems like types here are not right, 
could you try your code to see if TS complaints?

Reply:
Fixed types @BFEdev